tests/test_homo_sapiens.py

```diff
@@ -27,36 +28,77 @@ def test_str(self):
+        self.assertEqual(genome.chromosomes["chr4"].length, 191154276)
+        self.assertEqual(genome.chromosomes["chr5"].length, 180915260)
+        self.assertEqual(genome.chromosomes["chr6"].length, 171115067)
+        self.assertEqual(genome.chromosomes["chr7"].length, 159138663)
+        self.assertEqual(genome.chromosomes["chr8"].length, 146364022)
+        self.assertEqual(genome.chromosomes["chr9"].length, 141213431)
+        self.assertEqual(genome.chromosomes["chr10"].length, 135534747)
+        self.assertEqual(genome.chromosomes["chr11"].length, 135006516)
+        self.assertEqual(genome.chromosomes["chr12"].length, 133851895)
+        self.assertEqual(genome.chromosomes["chr13"].length, 115169878)
+        self.assertEqual(genome.chromosomes["chr14"].length, 107349540)
+        self.assertEqual(genome.chromosomes["chr15"].length, 102531392)
+        self.assertEqual(genome.chromosomes["chr16"].length, 90354753)
+        self.assertEqual(genome.chromosomes["chr17"].length, 81195210)
+        self.assertEqual(genome.chromosomes["chr18"].length, 78077248)
+        self.assertEqual(genome.chromosomes["chr19"].length, 59128983)
+        self.assertEqual(genome.chromosomes["chr20"].length, 63025520)
+        self.assertEqual(genome.chromosomes["chr21"].length, 48129895)
+        self.assertEqual(genome.chromosomes["chr22"].length, 51304566)
+        self.assertEqual(genome.chromosomes["chrX"].length, 155270560)
+        self.assertEqual(genome.chromosomes["chrY"].length, 59373566)
+
+    def get_mean_rr_numpy(self, chromosome):
+        chrom_length = chromosome.length
+        recombination_map = chromosome.recombination_map()
+        positions = np.array(recombination_map.get_positions())
+        positions_diff = recombination_map.get_positions()[1:]
+        positions_diff.append(chrom_length)
+        positions_diff = np.array(positions_diff)
+        window_sizes = positions_diff - positions
+        weights = window_sizes / chrom_length
+        rates = recombination_map.get_rates()
+        return np.average(rates, weights=weights)
+
+    def test_default_recombination_rates(self):
+        """
+        recompute default recombination rates from maps found at
+        "http://ftp-trace.ncbi.nih.gov/1000genomes/ftp/technical/working/"
+        "20110106_recombination_hotspots/ then compare the results to
+        the current default recombination rates for each chromosome"
+        """
+        # We catch the warning that will be thrown when we iterate over
+        # Y chromosome.
+        with self.assertWarns(Warning):
+            for chrom in homo_sapiens.genome.chromosomes.values():
+                default_rr = chrom.default_recombination_rate
+                numpy_rr = self.get_mean_rr_numpy(chrom)
+                # Assert that the difference in mean recombination rate
+                # is small when computed with numpy.
+                self.assertGreater(1e-11, abs(numpy_rr-default_rr))
```

Comment:
Use ``np.allclose`` here, comparing floating point numbers for approximate equality is subtle and tricky.

Reply:
Oh, interesting, totally makes sense. will do.